cmd/keys.go

```diff
@@ -244,6 +280,26 @@ $ %s k s eth test-key`, appName, appName)),
 	return cmd
 }
 
+// validateAddKeyInput checks that the AddKeyInput is valid.
+func validateAddKeyInput(input *AddKeyInput) error {
+	// validate remote signer address and URL if either is provided
+	if input.RemoteSigner.Address != "" || input.RemoteSigner.Url != "" {
+		if input.RemoteSigner.Address == "" {
+			return fmt.Errorf("remote signer address cannot be empty")
+		}
+		if input.RemoteSigner.Url == "" {
+			return fmt.Errorf("remote signer URL cannot be empty")
+		}
+	}
+
+	// ensure the user does not specify both a private key and a mnemonic
+	if input.PrivateKey != "" && input.Mnemonic != "" {
+		return fmt.Errorf("cannot specify both a private key and a mnemonic; please choose one")
+	}
+
+	return nil
+}
```

Comment:
don't we have to prevent input both private key and remove / mnemonic and remote?